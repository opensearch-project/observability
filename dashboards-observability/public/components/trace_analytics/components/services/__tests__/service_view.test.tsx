/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { ServiceView } from '..';
import { coreStartMock } from '../../../../../../test/__mocks__/coreMocks';

describe('Service view component', () => {
  configure({ adapter: new Adapter() });

  it('renders service view', () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const addFilter = jest.fn();
    const wrapper = shallow(
      <ServiceView
        serviceName="order"
        chrome={core.chrome}
        appConfigs={[]}
        parentBreadcrumbs={[{ text: 'test', href: 'test#/' }]}
        http={core.http}
        query=""
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        addFilter={addFilter}
        mode="data_prepper"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
