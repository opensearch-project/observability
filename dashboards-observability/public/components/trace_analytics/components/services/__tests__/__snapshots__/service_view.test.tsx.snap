// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service view component renders service view 1`] = `
<Fragment>
  <EuiPage>
    <EuiPageBody>
      <EuiFlexGroup
        alignItems="center"
        gutterSize="s"
      >
        <EuiFlexItem>
          <EuiTitle
            size="l"
          >
            <h2
              className="overview-content"
            >
              order
            </h2>
          </EuiTitle>
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiSuperDatePicker
            commonlyUsedRanges={
              Array [
                Object {
                  "end": "now/d",
                  "label": "Today",
                  "start": "now/d",
                },
                Object {
                  "end": "now/w",
                  "label": "This week",
                  "start": "now/w",
                },
                Object {
                  "end": "now/M",
                  "label": "This month",
                  "start": "now/M",
                },
                Object {
                  "end": "now/y",
                  "label": "This year",
                  "start": "now/y",
                },
                Object {
                  "end": "now-1d/d",
                  "label": "Yesterday",
                  "start": "now-1d/d",
                },
                Object {
                  "end": "now",
                  "label": "Week to date",
                  "start": "now/w",
                },
                Object {
                  "end": "now",
                  "label": "Month to date",
                  "start": "now/M",
                },
                Object {
                  "end": "now",
                  "label": "Year to date",
                  "start": "now/y",
                },
              ]
            }
            dateFormat="MMM D, YYYY @ HH:mm:ss.SSS"
            end="now"
            isAutoRefreshOnly={false}
            isDisabled={false}
            isPaused={true}
            onTimeChange={[Function]}
            recentlyUsedRanges={Array []}
            refreshInterval={0}
            showUpdateButton={false}
            start="now-5m"
            timeFormat="HH:mm"
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        />
      </EuiFlexGroup>
      <EuiSpacer
        size="xl"
      />
      <EuiPanel>
        <PanelTitle
          title="Overview"
        />
        <EuiHorizontalRule
          margin="m"
        />
        <EuiFlexGroup>
          <EuiFlexItem>
            <EuiFlexGroup
              direction="column"
            >
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Name
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  order
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Number of connected services
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  0
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Connected services
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiFlexGroup
              direction="column"
            >
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Average latency (ms)
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Error rate
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Throughput
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem
                grow={false}
              >
                <EuiText
                  className="overview-title"
                >
                  Traces
                </EuiText>
                <EuiText
                  className="overview-content"
                  size="s"
                >
                  -
                </EuiText>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer />
      </EuiPanel>
      <EuiSpacer />
      <ServiceMap
        currService="order"
        idSelected="latency"
        serviceMap={Object {}}
        setIdSelected={[Function]}
      />
      <EuiSpacer />
      <EuiPanel>
        <PanelTitle
          title="Spans"
          totalItems={0}
        />
        <EuiHorizontalRule
          margin="m"
        />
        <div>
          <SpanDetailTable
            DSL={Object {}}
            hiddenColumns={
              Array [
                "serviceName",
              ]
            }
            http={[MockFunction]}
            openFlyout={[Function]}
            setTotal={[Function]}
          />
        </div>
      </EuiPanel>
    </EuiPageBody>
  </EuiPage>
</Fragment>
`;
